examples/flutter_gallery/lib/gallery/app.dart

```diff
@@ -140,7 +135,67 @@ class _GalleryAppState extends State<GalleryApp> {
           child: _applyTextScaleFactor(child),
         );
       },
-      home: home,
+      home: FutureBuilder<bool>(
+        future: _checkWelcomeFuture,
+        builder: (BuildContext context, AsyncSnapshot<bool> snapshot) {
+          if (snapshot.connectionState == ConnectionState.done) {
+            _showWelcome ??= snapshot.data;
+            return _homeWidget(context);
+          } else {
+            return Container();
+          }
+        },
+      ),
     );
   }
+
+  Future<bool> _checkWelcomeState() async {
+    _prefs = await SharedPreferences.getInstance();
+    final bool hasSeenWelcome = _prefs.getBool('hasSeenWelcome') ?? false;
+    return !hasSeenWelcome;
+  }
+
+  Widget _homeWidget(BuildContext context) {
```

Comment:
Make the name match how much it costs. i.e. you're building when you're calling so call this buildHomeWidget.